cryptography/src/jvmMain/kotlin/com/wire/kalium/cryptography/ProteusClient.kt

```diff
@@ -0,0 +1,112 @@
+package com.wire.kalium.cryptography
+
+import com.wire.bots.cryptobox.CryptoBox
+import com.wire.bots.cryptobox.CryptoException
+import com.wire.kalium.cryptography.exceptions.ProteusException
+import java.util.ArrayList
+import java.util.Base64
+import java.util.UUID
+
+actual class ProteusClient {
+
+    val path: String
+    var box: CryptoBox? = null
+
+    actual constructor(rootDir: String, userId: String) {
+        path = String.format("%s/%s", rootDir, userId)
+    }
+
```

Comment:
is there a reason the constructor is not a primary constructor and the  box not null, , something like 
``` Kotlin
actual class ProteusClient (rootDir: String, userId: String) {

    private val path: String
    private lateinit var box: CryptoBox

    init {
        path = String.format("%s/%s", rootDir, userId)
    }

    actual fun open() {
        box = wrapException { CryptoBox.open(path) }
    }
```

Reply:
Wasn't aware the distinction :) I'll make it a primary constructor and I'll initialise the box there.

Reply:
Ah about initialising in the constructor I'm unsure how that will translate into Objective-C when the constructor can throw an exception.

But I'll the  `lateinit`

Reply:
You can annotate the constructor too:

```kotlin
actual class ProteusClient @Throws(ProteusException::class) constructor(rootDir: String, userId: String) {
```

If you want to annotate the primary constructor, the keyword `constructor` must be explicit like in the example above.

Reply:
done

and about throwing in constructor I would avoid since it's not very idiomatic to fail in constructors in Object-C and I don't know how that would be translated.

Reply:
if you need to add actual you can
```
actual class ProteusClient @Throws(ProteusException::class) actual constructor(rootDir: String, userId: String) {
```